firebaseai/FirebaseAIExample/Features/Live/Views/ConnectButton.swift

```diff
@@ -0,0 +1,100 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+struct ConnectButton: View {
+  var state: LiveViewModelState
+  var onConnect: () async -> Void
+  var onDisconnect: () async -> Void
+
+  @State private var gradientAngle: Angle = .zero
+
+  private var isConnected: Bool { state == .connected }
+
+  private var title: String {
+    switch state {
+    case .connected: "Stop"
+    case .connecting: "Connecting..."
+    case .idle: "Start"
+    }
+  }
+
+  private var image: String {
+    switch state {
+    case .connected: "stop.fill"
```

Comment:
Use the non-fill variants of the system icons to make the design look more at home on iOS.